### ISSUE TITLE
Use const over define

### ISSUE THREAD

**Issue by ScreamingDev** [kalessil/phpinspectionsea#1077]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Feature request                         |
| Plugin         | Php Inspections (EA Ultimate) |
| Language level | e.g. PHP 7.2 (check project settings or composer.json)                        |

Expected behaviour
--

```
define('FOO', 'bar');  // Hint: Use const here
define('BAR', 5); // Hint: Use const ffs
define('BAZ', 5+5); // Nothing
```

I've read some time ago that const are build during compile time and define is read during runtime.
So using const over define would allow to skip some steps during runtime
as the const is already given in the compiled cache.

This needs to be evaluated but I can't find the link. I guess it was on stack overflow.

**Comment by kalessil** [comment]
Will check, in which contexts this is relevant (global scopes, functions, methods)?

**Comment by ScreamingDev** [comment]
Only global scope. Like config files or bootstrapping things.
As soon as it is within a lower scope (function, for, if, while, ...) it looses all benefits.

**Comment by ScreamingDev** [comment]
In addition: This should be a suggestion not mandatory. Otherwise code would mix up like

```php
const PIE = 3;
define( 'WUB', random_int(-PIE,PIE) );
const HELLO = 'World!';
define( 'BYE', $x + $y * 2 );
```

And devs don't like ugly things no matter how much sense they make ;D I can tell.
